app/src/main/java/org/wikipedia/analytics/eventplatform/ImageRecommendationsEvent.kt

```diff
@@ -0,0 +1,60 @@
+
+    companion object {
+        private const val STREAM_NAME = "android.android_image_recommendation_event"
+        val reasons = listOf("notrelevant", "noinfo", "offensive", "lowquality", "unfamiliar", "other")
+        private val timer = ActiveTimer()
+
+        fun logImpression(activeInterface: String, actionData: String = "", wikiId: String = "") {
+            if (activeInterface == "recommendedimagetoolbar") {
+                timer.reset()
+            }
+            submitImageRecommendationEvent("impression", activeInterface, actionData, wikiId)
+        }
+
+        fun logAction(action: String, activeInterface: String, actionData: String = "", wikiId: String = "") {
+            submitImageRecommendationEvent(action, activeInterface, actionData, wikiId)
+        }
+
+        fun logSeEditSuccess(action: DescriptionEditActivity.Action, wikiId: String, revisionId: Long) {
+            when (action) {
+                DescriptionEditActivity.Action.ADD_DESCRIPTION -> logAction("edit_success", "se_add_description", getActionDataString(revisionId = revisionId.toString()), wikiId)
+                DescriptionEditActivity.Action.TRANSLATE_DESCRIPTION -> logAction("edit_success", "se_translate_description", getActionDataString(revisionId = revisionId.toString()), wikiId)
+                DescriptionEditActivity.Action.ADD_CAPTION -> logAction("edit_success", "se_add_caption", getActionDataString(revisionId = revisionId.toString()), wikiId)
+                DescriptionEditActivity.Action.TRANSLATE_CAPTION -> logAction("edit_success", "se_translate_caption", getActionDataString(revisionId = revisionId.toString()), wikiId)
+                else -> logAction("edit_success", "se_add_image_tags", getActionDataString(revisionId = revisionId.toString()), wikiId)
+            }
+        }
+
+        fun getActionDataString(filename: String? = null, recommendationSource: String? = null,
+                                recommendationSourceProjects: String? = null, rejectionReasons: String? = null,
+                                acceptanceState: String? = null, revisionId: String? = null, captionAdd: Boolean? = null,
+                                altTextAdd: Boolean? = null, addTimeSpent: Boolean = false): String {
+            return "${filename?.let { "filename: ${URLEncoder.encode(filename, "UTF-8")}, " }.orEmpty()}${recommendationSource?.let { "recommendation_source: $it, " }.orEmpty()}" +
+                    "${recommendationSourceProjects?.let { "recommendation_source_project: $it, " }.orEmpty()}${rejectionReasons?.let { "rejection_reasons: $it, " }.orEmpty()}" +
+                    "${acceptanceState?.let { "acceptance_state: $it, " }.orEmpty()}${revisionId?.let { "revision_id: $it, " }.orEmpty()}" +
+                    "${captionAdd?.let { "caption_add: $it, " }.orEmpty()}${altTextAdd?.let { "alt_text_add: $it, " }.orEmpty()}${if (addTimeSpent) "alt_text_add: " + timer.elapsedMillis.toString() else ""}"
+        }
```

Comment:
Would it be possible to have an extra `val` for the strings to increase the readability? 
```
val filenameStr = filename?.let { "filename: ${URLEncoder.encode(filename, "UTF-8")}, " }.orEmpty()
val recSourceStr = recommendationSource?.let { "recommendation_source: $it, " }.orEmpty()
return filenameStr + recSourceStr
```